src/plugins/usage_collection/server/usage_counters/usage_counters_service.ts

```diff
@@ -35,8 +37,8 @@ export interface UsageCountersServiceDeps {
 }
 
 export interface UsageCountersServiceSetup {
-  createUsageCounter: (type: string) => UsageCounter;
-  getUsageCounterByType: (type: string) => UsageCounter | undefined;
+  createUsageCounter: (domainId: string, soType?: UsageCounterSavedObjectType) => IUsageCounter;
```

Comment:
Maybe, instead of `soType`, we can add `source?: 'ui' | 'server'`, and let the `soType` be an implementation detail?

Reply:
Fair point! I'll update this